Build failure: oculante

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

oculante fails to build on unstable, probably related https://github.com/NixOS/nixpkgs/issues/475479

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/318072676

### Relevant log output

```console
nix log /nix/store/vr3y652a7pa1ad6vc2pp39r5asa026ff-oculante-0.9.2.1-unstable-2025-10-08.drv
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/i1m5k0lalyg0306m5w62y28654ydz9x2-source
source root is source
Executing cargoSetupPostUnpackHook
Finished cargoSetupPostUnpackHook
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
Executing cargoSetupPostPatchHook
Validating consistency between /build/source/Cargo.lock and /build/oculante-0.9.2.1-unstable-2025-10-08-vendor/Cargo.lock
Finished cargoSetupPostPatchHook
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
Executing cargoBuildHook
cargoBuildHook flags: -j 32 --target x86_64-unknown-linux-gnu --offline --profile release
   Compiling proc-macro2 v1.0.95
   Compiling unicode-ident v1.0.18
   Compiling autocfg v1.4.0
   Compiling cfg-if v1.0.0
   Compiling libc v0.2.172
   Compiling once_cell v1.21.3
   Compiling shlex v1.3.0
   Compiling libm v0.2.11
   Compiling pkg-config v0.3.32
   Compiling smallvec v1.15.0
   Compiling value-bag v1.11.1
   Compiling crossbeam-utils v0.8.21
   Compiling pin-project-lite v0.2.16
   Compiling memchr v2.7.4
   Compiling either v1.15.0
   Compiling version_check v0.9.5
   Compiling serde v1.0.219
   Compiling rayon-core v1.12.1
   Compiling bitflags v2.9.0
   Compiling futures-core v0.3.31
   Compiling arrayvec v0.7.6
   Compiling cfg_aliases v0.1.1
   Compiling bitflags v1.3.2
   Compiling simd-adler32 v0.3.7
   Compiling futures-io v0.3.31
   Compiling zerocopy v0.8.24
   Compiling adler2 v2.0.0
   Compiling thiserror v1.0.69
   Compiling foldhash v0.1.5
   Compiling byteorder v1.5.0
   Compiling crc32fast v1.4.2
   Compiling lazy_static v1.5.0
   Compiling xml-rs v0.8.26
   Compiling equivalent v1.0.2
   Compiling scopeguard v1.2.0
   Compiling parking_lot_core v0.9.10
   Compiling libloading v0.8.6
   Compiling stable_deref_trait v1.2.0
   Compiling anyhow v1.0.98
   Compiling tracing-core v0.1.33
   Compiling miniz_oxide v0.8.8
   Compiling hashbrown v0.15.2
   Compiling log v0.4.27
   Compiling fastrand v2.3.0
   Compiling ttf-parser v0.25.1
   Compiling rustix v0.38.44
   Compiling futures-sink v0.3.31
   Compiling pin-utils v0.1.0
   Compiling downcast-rs v1.2.1
   Compiling ab_glyph_rasterizer v0.1.8
   Compiling linux-raw-sys v0.4.15
   Compiling futures-channel v0.3.31
   Compiling fdeflate v0.3.7
   Compiling futures-task v0.3.31
   Compiling ahash v0.8.11
   Compiling typenum v1.18.0
   Compiling zerocopy v0.7.35
   Compiling parking v2.2.1
   Compiling syn v1.0.109
   Compiling paste v1.0.15
   Compiling num-traits v0.2.19
   Compiling slab v0.4.9
   Compiling lock_api v0.4.12
   Compiling winnow v0.7.6
   Compiling unicase v2.8.1
   Compiling percent-encoding v2.3.1
   Compiling memoffset v0.6.5
   Compiling dlib v0.5.2
   Compiling xmlparser v0.13.6
   Compiling icu_locid_transform_data v1.5.1
   Compiling color_quant v1.1.0
   Compiling writeable v0.5.5
   Compiling minimal-lexical v0.2.1
   Compiling quick-error v2.0.1
   Compiling litemap v0.7.5
   Compiling itoa v1.0.15
   Compiling imgref v1.11.0
   Compiling crossbeam-epoch v0.9.18
   Compiling quote v1.0.40
   Compiling flate2 v1.1.1
   Compiling concurrent-queue v2.5.0
   Compiling nom v7.1.3
   Compiling icu_properties_data v1.5.1
   Compiling indexmap v2.9.0
   Compiling roxmltree v0.14.1
   Compiling built v0.7.7
   Compiling fnv v1.0.7
   Compiling weezl v0.1.8
   Compiling khronos_api v3.1.0
   Compiling aligned-vec v0.5.0
   Compiling mint v0.5.9
   Compiling icu_normalizer_data v1.5.1
   Compiling syn v2.0.100
   Compiling jobserver v0.1.33
   Compiling wayland-scanner v0.29.5
   Compiling event-listener v5.4.0
   Compiling crossbeam-deque v0.8.6
   Compiling wayland-sys v0.29.5
   Compiling slotmap v1.0.7
   Compiling instant v0.1.13
   Compiling heck v0.5.0
   Compiling byteorder-lite v0.1.0
   Compiling rustversion v1.0.20
   Compiling png v0.17.16
   Compiling float-cmp v0.9.0
   Compiling gif v0.13.1
   Compiling cc v1.2.19
   Compiling event-listener-strategy v0.5.4
   Compiling rav1e v0.7.1
   Compiling strict-num v0.1.1
   Compiling getrandom v0.2.15
   Compiling simd_helpers v0.1.0
   Compiling futures-lite v2.6.0
   Compiling generic-array v0.14.7
   Compiling itertools v0.12.1
   Compiling zune-inflate v0.2.54
   Compiling notan_macro v0.12.1
   Compiling bitstream-io v2.6.0
   Compiling utf8_iter v1.0.4
   Compiling new_debug_unreachable v1.0.6
   Compiling bytes v1.10.1
   Compiling utf16_iter v1.0.5
   Compiling write16 v1.0.0
   Compiling zune-core v0.4.12
   Compiling num-integer v0.1.46
   Compiling half v2.6.0
   Compiling approx v0.5.1
   Compiling v_frame v0.3.8
   Compiling num-complex v0.4.6
   Compiling noop_proc_macro v0.3.0
   Compiling arrayref v0.3.9
   Compiling zune-jpeg v0.4.14
   Compiling rand_core v0.6.4
   Compiling gl_generator v0.14.0
   Compiling loop9 v0.1.5
   Compiling rayon v1.10.0
   Compiling num-bigint v0.4.6
   Compiling num-iter v0.1.45
   Compiling wayland-sys v0.31.6
   Compiling avif-serialize v0.8.3
   Compiling semver v1.0.26
   Compiling cmake v0.1.54
   Compiling nix v0.24.3
   Compiling parking_lot v0.12.3
   Compiling lebe v0.5.2
   Compiling ppv-lite86 v0.2.21
   Compiling allocator-api2 v0.2.21
   Compiling siphasher v1.0.1
   Compiling encoding_index_tests v0.1.4
   Compiling wayland-client v0.29.5
   Compiling bit_field v0.10.2
   Compiling getrandom v0.3.2
   Compiling image-webp v0.2.1
   Compiling async-lock v3.4.0
   Compiling notan_utils v0.12.1
   Compiling exr v1.73.0
   Compiling x11-dl v2.21.0
   Compiling phf_shared v0.11.3
   Compiling form_urlencoded v1.2.1
   Compiling aho-corasick v1.1.3
   Compiling static_assertions v1.1.0
   Compiling rustix v1.0.5
   Compiling subtle v2.6.1
   Compiling regex-syntax v0.8.5
   Compiling scoped-tls v1.0.1
   Compiling async-task v4.7.1
   Compiling wayland-protocols v0.29.5
   Compiling rand_chacha v0.3.1
   Compiling async-channel v2.3.1
   Compiling quick-xml v0.37.4
   Compiling itertools v0.13.0
   Compiling wayland-client v0.31.8
   Compiling linux-raw-sys v0.9.4
   Compiling notan_core v0.12.1
   Compiling iana-time-zone v0.1.63
   Compiling owned_ttf_parser v0.25.0
   Compiling serde_json v1.0.140
   Compiling rand v0.8.5
   Compiling ryu v1.0.20
   Compiling atomic-waker v1.1.2
   Compiling ab_glyph v0.2.29
   Compiling chrono v0.4.40
   Compiling piper v0.2.4
   Compiling socket2 v0.5.9
   Compiling shaderc-sys v0.8.3
   Compiling spirv_cross v0.23.1
   Compiling wayland-backend v0.3.8
   Compiling ring v0.17.14
   Compiling num-rational v0.4.2
   Compiling mio v1.0.3
   Compiling nix v0.25.1
   Compiling smithay-client-toolkit v0.16.1
   Compiling xcursor v0.3.8
   Compiling wasm-bindgen-shared v0.2.100
   Compiling nohash-hasher v0.2.0
   Compiling rustls-pki-types v1.11.0
   Compiling shaderc v0.8.3
   Compiling num v0.4.3
   Compiling wayland-scanner v0.31.6
   Compiling vec_map v0.8.2
   Compiling mime v0.3.17
   Compiling tokio v1.44.2
   Compiling wayland-commons v0.29.5
   Compiling blocking v1.6.1
   Compiling crypto-common v0.1.6
   Compiling glutin_glx_sys v0.4.0
   Compiling glutin_egl_sys v0.5.1
   Compiling av1-grain v0.2.3
   Compiling http v1.3.1
   Compiling memmap2 v0.5.10
   Compiling euclid v0.22.11
   Compiling wayland-sys v0.30.1
   Compiling regex-automata v0.4.9
   Compiling untrusted v0.9.0
   Compiling jpeg-decoder v0.3.1
   Compiling maybe-rayon v0.1.1
   Compiling safe-transmute v0.11.3
   Compiling fixedbitset v0.4.2
   Compiling httparse v1.10.1
   Compiling thiserror v2.0.12
   Compiling petgraph v0.6.5
   Compiling twox-hash v2.1.0
   Compiling zstd-sys v2.0.15+zstd.1.5.7
   Compiling encoding-index-japanese v1.20141219.5
   Compiling encoding-index-korean v1.20141219.5
   Compiling synstructure v0.13.1
   Compiling encoding-index-tradchinese v1.20141219.5
   Compiling encoding-index-singlebyte v1.20141219.5
   Compiling encoding-index-simpchinese v1.20141219.5
   Compiling tempfile v3.19.1
   Compiling http-body v1.0.1
   Compiling winit v0.28.7
   Compiling crossbeam-channel v0.5.15
   Compiling lyon_geom v1.0.6
   Compiling memoffset v0.9.1
   Compiling glutin v0.30.10
   Compiling option-ext v0.2.0
   Compiling endi v1.1.0
   Compiling rustc-hash v2.1.1
   Compiling tiff v0.9.1
   Compiling bumpalo v3.17.0
   Compiling rustls v0.23.26
   Compiling linked-hash-map v0.5.6
   Compiling crevice_notan-derive v0.13.0
   Compiling cfg_aliases v0.2.1
   Compiling termcolor v1.4.1
   Compiling try-lock v0.2.5
   Compiling xi-unicode v0.3.0
   Compiling raw-window-handle v0.5.2
   Compiling wayland-cursor v0.29.5
   Compiling want v0.3.1
   Compiling lyon_path v1.0.7
   Compiling glyph_brush_layout v0.2.4
   Compiling nix v0.29.0
   Compiling glyph_brush_draw_cache v0.1.6
   Compiling wasm-bindgen-backend v0.2.100
   Compiling encoding v0.2.33
   Compiling rustc_version v0.4.1
   Compiling block-buffer v0.10.4
   Compiling async-executor v1.13.1
   Compiling mio v0.8.11
   Compiling os_pipe v1.2.1
   Compiling signal-hook-registry v1.4.2
   Compiling ordered-float v5.0.0
   Compiling core_maths v0.1.1
   Compiling mime_guess v2.0.5
   Compiling matrixmultiply v0.3.9
   Compiling byteordered v0.6.0
   Compiling glutin-winit v0.3.0
   Compiling naga v22.1.0
   Compiling libloading v0.7.4
   Compiling khronos-egl v6.0.0
   Compiling az v1.2.1
   Compiling tower-service v0.3.3
   Compiling dunce v1.0.5
   Compiling bit-vec v0.7.0
   Compiling utf8parse v0.2.2
   Compiling unicode-width v0.1.14
   Compiling proc-macro-hack-impl v0.4.3
   Compiling x11rb-protocol v0.13.1
   Compiling ash v0.38.0+1.3.281
   Compiling siphasher v0.3.11
   Compiling proc-macro-hack v0.5.20+deprecated
   Compiling tree_magic_mini v3.1.6
   Compiling anstyle-parse v0.2.6
   Compiling lcms2-sys v4.0.5
   Compiling bit-set v0.6.0
   Compiling glyph_brush v0.7.12
   Compiling phf_shared v0.10.0
   Compiling codespan-reporting v0.11.1
   Compiling ttf-parser v0.24.1
   Compiling pathfinder_simd v0.5.4
   Compiling regex v1.11.1
   Compiling wasm-bindgen-macro-support v0.2.100
   Compiling digest v0.10.7
   Compiling serde_derive v1.0.219
   Compiling bytemuck_derive v1.9.3
   Compiling tracing-attributes v0.1.28
   Compiling thiserror-impl v1.0.69
   Compiling zerofrom-derive v0.1.6
   Compiling yoke-derive v0.7.5
   Compiling futures-macro v0.3.31
   Compiling zerovec-derive v0.10.3
   Compiling displaydoc v0.2.5
   Compiling profiling-procmacros v1.0.16
   Compiling icu_provider_macros v1.5.0
   Compiling num-derive v0.4.2
   Compiling arg_enum_proc_macro v0.3.4
   Compiling enumflags2_derive v0.7.11
   Compiling profiling v1.0.16
   Compiling snafu-derive v0.8.5
   Compiling zeroize_derive v1.4.2
   Compiling thiserror-impl v2.0.12
   Compiling futures-util v0.3.31
   Compiling phf_generator v0.11.3
   Compiling wayland-protocols v0.32.6
   Compiling bytemuck v1.22.0
   Compiling bzip2-sys v0.1.13+1.0.8
   Compiling lzma-sys v0.1.20
   Compiling freetype-sys v0.20.1
   Compiling libaom-sys v0.17.2+libaom.3.11.0
   Compiling phf v0.11.3
   Compiling generator v0.8.4
   Compiling rgb v0.8.50
   Compiling glam v0.24.2
   Compiling qoi v0.4.1
   Compiling calloop v0.10.6
   Compiling safe_arch v0.7.4
   Compiling tracing v0.1.41
   Compiling zerofrom v0.1.6
   Compiling tiny-skia-path v0.8.4
   Compiling yoke v0.7.5
   Compiling zeroize v1.8.1
   Compiling image v0.24.9
   Compiling yeslogic-fontconfig-sys v6.0.0
   Compiling spirv v0.3.0+sdk-1.3.268.0
   Compiling gpu-alloc-types v0.3.0
   Compiling zerovec v0.10.4
   Compiling polling v3.7.4
   Compiling jxl-bitstream v0.4.1
   Compiling jxl-threadpool v0.1.2
   Compiling jxl-grid v0.4.2
   Compiling wide v0.7.32
   Compiling gpu-descriptor-types v0.2.0
   Compiling wgpu-hal v22.0.0
   Compiling async-io v2.4.0
   Compiling jxl-coding v0.4.1
   Compiling tiny-skia v0.8.4
   Compiling anstyle v1.0.10
   Compiling rawpointer v0.2.1
   Compiling litrs v0.4.1
   Compiling doc-comment v0.3.3
   Compiling colorchoice v1.0.3
   Compiling unicode-xid v0.2.6
   Compiling radium v0.7.0
   Compiling glow v0.12.3
   Compiling zstd-safe v7.2.4
   Compiling tinyvec_macros v0.1.1
   Compiling jxl-color v0.7.1
   Compiling jxl-modular v0.7.1
   Compiling float_next_after v1.0.0
   Compiling owo-colors v4.2.0
   Compiling hexf-parse v0.2.1
   Compiling roxmltree v0.20.0
   Compiling async-signal v0.2.10
   Compiling portable-atomic v1.11.0
   Compiling adler32 v1.2.0
   Compiling is_terminal_polyfill v1.70.1
   Compiling tinystr v0.7.6
   Compiling icu_collections v1.5.0
   Compiling fixed v1.29.0
   Compiling rustc-hash v1.1.0
   Compiling fast-srgb8 v1.0.0
   Compiling icu_locid v1.5.0
   Compiling anstyle-query v1.1.2
   Compiling base64 v0.22.1
   Compiling wasm-bindgen v0.2.100
   Compiling snafu v0.8.5
   Compiling x11rb v0.13.1
   Compiling jxl-image v0.9.0
   Compiling dicom-core v0.8.1
   Compiling jxl-vardct v0.7.0
   Compiling icu_provider v1.5.0
   Compiling anstream v0.6.18
   Compiling fontconfig-parser v0.5.7
   Compiling async-process v2.3.0
   Compiling notan_math v0.12.1
   Compiling crevice_notan v0.13.0
   Compiling icu_locid_transform v1.5.0
   Compiling document-features v0.2.11
   Compiling lyon_tessellation v1.0.15
   Compiling tinyvec v1.9.0
   Compiling gpu-descriptor v0.3.1
   Compiling futures-executor v0.3.31
   Compiling platter2 v0.1.6
   Compiling sctk-adwaita v0.5.4
   Compiling hyper v1.6.0
   Compiling gpu-alloc v0.6.0
   Compiling futures v0.3.31
   Compiling mime_guess2 v2.3.0
   Compiling tiny-skia-path v0.11.4
   Compiling icu_properties v1.5.1
   Compiling wayland-protocols-wlr v0.3.6
   Compiling phf_macros v0.11.3
   Compiling wasm-bindgen-macro v0.2.100
   Compiling env_filter v0.1.3
   Compiling dicom-dictionary-std v0.8.0
   Compiling turbojpeg-sys v1.1.1
   Compiling hyper-util v0.1.11
   Compiling serde_spanned v0.6.8
   Compiling toml_datetime v0.6.8
   Compiling zvariant_utils v3.2.0
   Compiling emath v0.27.2
   Compiling ecolor v0.27.2
   Compiling toml_edit v0.22.24
   Compiling enumflags2 v0.7.11
   Compiling hashbrown v0.14.5
   Compiling wl-clipboard-rs v0.9.2
error: failed to run custom build command for `libaom-sys v0.17.2+libaom.3.11.0`

Caused by:
  process didn't exit successfully: `/build/source/target/release/build/libaom-sys-c5d9470a0ec04129/build-script-build` (exit status: 101)
  --- stdout
  CMAKE_TOOLCHAIN_FILE_x86_64-unknown-linux-gnu = None
  CMAKE_TOOLCHAIN_FILE_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_TOOLCHAIN_FILE = None
  CMAKE_TOOLCHAIN_FILE = None
  CMAKE_GENERATOR_x86_64-unknown-linux-gnu = None
  CMAKE_GENERATOR_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_GENERATOR = None
  CMAKE_GENERATOR = None
  CMAKE_PREFIX_PATH_x86_64-unknown-linux-gnu = None
  CMAKE_PREFIX_PATH_x86_64_unknown_linux_gnu = None
  HOST_CMAKE_PREFIX_PATH = None
  CMAKE_PREFIX_PATH = None
  CMAKE_x86_64-unknown-linux-gnu = None
  CMAKE_x86_64_unknown_linux_gnu = None
  HOST_CMAKE = None
  CMAKE = None
  running: cd "/build/source/target/x86_64-unknown-linux-gnu/release/build/libaom-sys-c3b7272d33d3ab0f/out/build" && CMAKE_PREFIX_PATH="" LC_ALL="C" "cmake" "/build/oculante-0.9.2.1-unstable-2025-10-08-vendor/libaom-sys-0.17.2+libaom.3.11.0/vendor" "-DENABLE_DO>
  -- The C compiler identification is GNU 15.2.0
  -- The CXX compiler identification is GNU 15.2.0
  -- Detecting C compiler ABI info
  -- Detecting C compiler ABI info - done
  -- Check for working C compiler: /nix/store/r9wbjib6xxjkyb9yvjvrkl4sq61i2lyn-gcc-wrapper-15.2.0/bin/cc - skipped
  -- Detecting C compile features
  -- Detecting C compile features - done
  -- Detecting CXX compiler ABI info
  -- Detecting CXX compiler ABI info - done
  -- Check for working CXX compiler: /nix/store/r9wbjib6xxjkyb9yvjvrkl4sq61i2lyn-gcc-wrapper-15.2.0/bin/c++ - skipped
  -- Detecting CXX compile features
  -- Detecting CXX compile features - done
  -- Performing Test CMAKE_HAVE_LIBC_PTHREAD
  -- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
  -- Found Threads: TRUE
  -- Performing Test C_TEST_COMPILED
  -- Performing Test C_TEST_COMPILED - Failed
  -- The ASM_NASM compiler identification is NASM
  -- Found assembler: /nix/store/lywj07fkb4grpp6g7xmq49hg1d2nfsa0-nasm-3.01/bin/nasm
  -- Configuring incomplete, errors occurred!

  --- stderr
  CMake Deprecation Warning at CMakeLists.txt:14 (cmake_minimum_required):
    Compatibility with CMake < 3.10 will be removed from a future version of
    CMake.

    Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
    to tell CMake that the project requires at least <min> but has been updated
    to work with policies introduced by <max> or earlier.


  --- aom_configure: Detected CPU: x86_64
  Running C compiler test: pie_check
  C Compiler test pie_check failed.
  CMake Error at build/cmake/aom_optimization.cmake:219 (message):
    Unsupported nasm: multipass optimization not supported.
  Call Stack (most recent call first):
    build/cmake/aom_configure.cmake:170 (test_nasm)
    CMakeLists.txt:73 (include)



  thread 'main' (6094) panicked at /build/oculante-0.9.2.1-unstable-2025-10-08-vendor/cmake-0.1.54/src/lib.rs:1119:5:

  command did not execute successfully, got: exit status: 1

  build script failed, must exit now
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.63, NixOS, 26.05 (Yarara), 26.05.20251228.c0b0e0f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2+1`
 - nixpkgs: `/nix/store/xzl5dhddgjxp2p18h5qpkhvlkp8128j1-source`



### Notify maintainers

@dit7ya 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/oc/oculante/libaom-sys-0.17.2+libaom.3.11.0-cmake-nasm-fix.patch
+++ b/pkgs/by-name/oc/oculante/libaom-sys-0.17.2+libaom.3.11.0-cmake-nasm-fix.patch
@@ -1,0 +1,21 @@
+--- a/libaom-sys-0.17.2+libaom.3.11.0/vendor/build/cmake/aom_optimization.cmake
++++ b/libaom-sys-0.17.2+libaom.3.11.0/vendor/build/cmake/aom_optimization.cmake
+@@ -212,7 +212,7 @@ endfunction()
+ # Currently checks only for presence of required object formats and support for
+ # the -Ox argument (multipass optimization).
+ function(test_nasm)
+-  execute_process(COMMAND ${CMAKE_ASM_NASM_COMPILER} -hf
++  execute_process(COMMAND ${CMAKE_ASM_NASM_COMPILER} -hO
+                   OUTPUT_VARIABLE nasm_helptext)
+ 
+   if(NOT "${nasm_helptext}" MATCHES "-Ox")
+@@ -220,6 +220,9 @@ function(test_nasm)
+       FATAL_ERROR "Unsupported nasm: multipass optimization not supported.")
+   endif()
+ 
++  execute_process(COMMAND ${CMAKE_ASM_NASM_COMPILER} -hf
++                  OUTPUT_VARIABLE nasm_helptext)
++
+   if("${AOM_TARGET_CPU}" STREQUAL "x86")
+     if("${AOM_TARGET_SYSTEM}" STREQUAL "Darwin")
+       if(NOT "${nasm_helptext}" MATCHES "macho32")

--- a/pkgs/by-name/oc/oculante/package.nix
+++ b/pkgs/by-name/oc/oculante/package.nix
@@ -67,6 +67,18 @@ rustPlatform.buildRustPackage rec {
     "--skip=thumbnails::test_thumbs" # broken as of v0.9.2
   ];
 
+  patches = [
+    # The below patch is needed to fix this build, until the upstream dependency (libavif-rs) fixes the problem.
+    # The explicit `patchFlags` can also be removed when this patch becomes obsolete.
+    # <https://github.com/njaard/libavif-rs/issues/122>
+    ./libaom-sys-0.17.2+libaom.3.11.0-cmake-nasm-fix.patch
+  ];
+
+  patchFlags = [
+    "-p1"
+    "--directory=../${pname}-${version}-vendor"
+  ];
+
   postInstall = ''
     install -Dm444 $src/res/icons/icon.png $out/share/icons/hicolor/128x128/apps/oculante.png
     install -Dm444 $src/res/oculante.desktop -t $out/share/applications

